### PR TITLE
Set up app layout

### PR DESCRIPTION
**Pull request by bklaing2**
## Purpose
Set up the layout for the app router

closes: #332 

## Approach
Migrated the `Layout` component being used across the `pages` router to work with the `app` router. It now lives in the `src/app/layout.tsx` file, and any sibling or child `page.tsx` files are passed to it as children.

#### Open Questions and Pre-Merge TODOs
- [ ] The most significant changes are in the new `app` folder
  - Apparently when adopting the app router, NextJS automatically inserts a new TypeScript plugin on build. There are a lot of new linting error fixes included in this PR.
- [ ] I had to add a temporary `404.tsx` file in the `pages` directory because the 404 page was throwing an error.
  - I think it has to do with the current mixture of `pages` and `app` router
  - This should be able to be removed once everything is migrated to the `app` router

## Learning
Referenced the [NextJS docs](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts) to understand the app router.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      <head>
+        <link
+          href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,400i,600"
+          rel="stylesheet"
+        />
+        <link
+          rel="stylesheet"
+          href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1/css/academicons.min.css"
+        />
+        {/* <link
```
Comment (jrhoads) [inline src/app/layout.tsx:48]: Commented code

File: src/utils/server/session.ts
[the remark this answers is not in the document]
Comment (bklaing2) [reply src/utils/server/session.ts:1]: At the moment they're the same. During the upgrade process it's possible that I'll need to update anything that uses process.env, and I don't want to break any other areas of the code during the upgrade process. If you have any suggestion for this I'd be open to hear

[the remark this answers is not in the document]
Comment (jrhoads) [reply src/utils/server/session.ts:1]: I'm fine with this as a first step.

**Review by jrhoads — APPROVED** [review]
This is a good start.  We can incrementally replace the routes since those in the app router/directory take precedence over those in the pages directory.

**Pull request merged**